dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_RESIZE_EXPERIMENTAL_RESIZE_OP_IMPL_CVCUDA_H_
+#define DALI_OPERATORS_IMAGE_RESIZE_EXPERIMENTAL_RESIZE_OP_IMPL_CVCUDA_H_
+
+#include <cvcuda/OpHQResize.hpp>
+
+#include <vector>
+
+#include "dali/kernels/imgproc/resample/params.h"
+#include "dali/operators/image/resize/resize_op_impl.h"
+#include "dali/operators/nvcvop/nvcvop.h"
+
+namespace dali {
+
+template <int spatial_ndim>
+class ResizeOpImplCvCuda : public ResizeBase<GPUBackend>::Impl {
+ public:
+  explicit ResizeOpImplCvCuda(int minibatch_size) : minibatch_size_(minibatch_size) {}
+
+  static_assert(spatial_ndim == 2 || spatial_ndim == 3, "Only 2D and 3D resizing is supported");
+
+
+  /// Dimensionality of each separate frame. If input contains no channel dimension, one is added
+  static constexpr int frame_ndim = spatial_ndim + 1;
+
+  void Setup(TensorListShape<> &out_shape, const TensorListShape<> &in_shape, int first_spatial_dim,
+             span<const kernels::ResamplingParams> params) override {
+    // Calculate output shape of the input, as supplied (sequences, planar images, etc)
+    GetResizedShape(out_shape, in_shape, params, spatial_ndim, first_spatial_dim);
+
+    // Create "frames" from outer dimensions and "channels" from inner dimensions.
+    GetFrameShapesAndParams<spatial_ndim>(in_shape_, params_, in_shape, params, first_spatial_dim);
+
+    // Now that we have per-frame parameters, we can calculate the output shape of the
+    // effective frames (from videos, channel planes, etc).
+    GetResizedShape(out_shape_, in_shape_, make_cspan(params_), 0);
+
+    curr_minibatch_size_ = minibatch_size_;
+    // CV-CUDA tensors can't represent empty samples,
+    // so curr_minibatch_size_ is set to 1 to skip each empty sample one by one
+    if (HasEmptySamples(in_shape_)) {
+      curr_minibatch_size_ = 1;
+    }
```

Comment:
Why? Can't we just skip them when assembling the input TensorBatch?

Reply:
Empty samples in the input are a very much edge case, I assume, so it wouldn't affect regular execution. Nevertheless, I implemented the solution with global index array